### PR TITLE
add stm32f1 usb cdc/msc USB composite device

### PR DESCRIPTION
**Pull request by bigtreetech**
### Requirements
1. this PR need more test, and it does not deal with the situation that Marlin and PC access the same SD card at the same time, this need TODO
2. you must reinstall the maple USB driver in here https://github.com/bigtreetech/Marlin/tree/sync-stm32f1/Marlin/src/HAL/HAL_STM32F1/maple_win_usb_driver

### Description

1. Implement  USB composite device of CDC and mass storage, you can see the onboard sd card on PC by usb cable, just like this
image

### Related Issues
https://github.com/bigtreetech/BIGTREETECH-SKR-mini-E3-/issues/14

### CODE REVIEW

File: Marlin/src/HAL/HAL_STM32F1/msc_sd.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * (at your option) any later version.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "../../inc/MarlinConfig.h"
+#include "msc_sd.h"
+#include "onboard_sd.h"
+#include "spi.h"
```
Comment (tpruvot) [inline Marlin/src/HAL/HAL_STM32F1/msc_sd.cpp]: On linux, headers are case sensitive... should be SPI.h

but why SPI for USB stuff ?

**Comment by thinkyhead** [comment]
Please merge PR bigtreetech/Marlin#5 as soon as you get a chance. Thanks!

File: Marlin/src/HAL/HAL_STM32F1/msc_sd.cpp (hunk 1 above)
Comment (bigtreetech) [reply Marlin/src/HAL/HAL_STM32F1/msc_sd.cpp]: It has been cleaned up.

**Comment by bigtreetech** [comment]
> Please merge PR [bigtreetech#5](https://github.com/bigtreetech/Marlin/pull/5) as soon as you get a chance. Thanks!

Merged.

**Comment by thinkyhead** [comment]
One more patch ought to do it. This one has been tested and passes here. bigtreetech/Marlin#6

**Comment by thinkyhead** [comment]
Merged.

**Comment by reloxx13** [comment]
👍 
grafik

**Pull request closed without merging**